batch/2017-01-01.4.0/swagger/BatchService.json

```diff
@@ -0,0 +1,14081 @@
+            "x-ms-parameter-grouping": {
+              "postfix": "Options"
+            },
+            "x-ms-client-request-id": true
+          },
+          {
+            "name": "return-client-request-id",
+            "default": false,
+            "in": "header",
+            "required": false,
+            "type": "boolean",
+            "description": "Whether the server should return the client-request-id in the response.",
+            "x-ms-parameter-grouping": {
+              "postfix": "Options"
+            }
+          },
+          {
+            "name": "ocp-date",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "format": "date-time-rfc1123",
+            "description": "The time the request was issued. Client libraries typically set this to the current system clock time; set it explicitly if you are calling the REST API directly.",
+            "x-ms-parameter-grouping": {
+              "postfix": "Options"
+            }
+          },
+          {
+            "name": "ocp-range",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "description": "The byte range to be retrieved. The default is to retrieve the entire file. The format is bytes=startRange-endRange.",
+            "x-ms-parameter-grouping": {
+              "postfix": "Options"
+            }
+          },
+          {
+            "name": "If-Modified-Since",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "format": "date-time-rfc1123",
+            "description": "A timestamp indicating the last modified time of the resource known to the client. The operation will be performed only if the resource on the service has been modified since the specified time.",
+            "x-ms-parameter-grouping": {
+              "postfix": "Options"
+            }
+          },
+          {
+            "name": "If-Unmodified-Since",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "format": "date-time-rfc1123",
+            "description": "A timestamp indicating the last modified time of the resource known to the client. The operation will be performed only if the resource on the service has not been modified since the specified time.",
+            "x-ms-parameter-grouping": {
+              "postfix": "Options"
+            }
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "headers": {
+              "client-request-id": {
+                "description": "The client-request-id provided by the client during the request. This will be returned only if the return-client-request-id parameter was set to t
```

Comment:
@olydis Our server guarantees that it will not return unknown Enum values to clients that won't understand them. It does this using the `api-version` provided in the URI. In this particular Enum's case, it actually should never have had `Active` state in it to begin with as the server never returned that value. It was a mistake made when authoring the Swagger specification, so there is no risk to past clients.

You can read my comment [here](https://github.com/Azure/autorest/issues/1591#issuecomment-260433709) for more into what exactly the Batch service does and where it would be useful to use `modelAsString` (answer: It might be useful on enum's which are not "State" enums).

Still, I would be _really_ hesitant to make the switch to `modelAsString` today since the current experience for `modelAsString` is so poor and in the past 2 years we haven't run into any issues with it since the non-state enums in our service are pretty much static. If we do make a change to a non-state enum today, we will work around it on the server using `api-version` to avoid breaking clients while still maintaining a good client experience.

Once you guys have support for [1591](https://github.com/Azure/autorest/issues/1591), which was also asked for by me in [733](https://github.com/Azure/autorest/issues/733) and by @brjohnstmsft in [290](https://github.com/Azure/autorest/issues/290) then we will move to that.